src/Interval.php

```diff
@@ -40,6 +40,17 @@ public function __construct(Instant $startInclusive, Instant $endExclusive)
         $this->end = $endExclusive;
     }
 
+    /**
+     * @param Instant $startInclusive The start instant, inclusive.
+     * @param Instant $endExclusive   The end instant, exclusive.
+     *
+     * @throws DateTimeException If the end instant is before the start instant.
+     */
+    public static function of(Instant $startInclusive, Instant $endExclusive): Interval
```

Comment:
Maybe we should deprecate the public constructor in favour of this factory method, to keep it consistent with other classes?

Reply:
Yeah, it makes sense